### ISSUE TITLE
Type of constant not recognizes

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#3523]
I report it here because i dont see what could be wrong with the typesignature in stdlib

**Describe the bug**
Excerp from stdlib/subprocess.pyi

```
if sys.platform == "win32":
    _ENV: TypeAlias = Mapping[str, str]
else:
    _ENV: TypeAlias = Mapping[bytes, StrOrBytesPath] | Mapping[str, StrOrBytesPath]

_T = TypeVar("_T")

# These two are private but documented
if sys.version_info >= (3, 11):
    _USE_VFORK: bool
if sys.version_info >= (3, 8):
    _USE_POSIX_SPAWN: bool

class CompletedProcess(Generic[_T]):
    # morally: _CMD
    args: Any
    returncode: int
    # These can both be None, but requiring checks for None would be tedious
    # and writing all the overloads would be horrific.
    stdout: _T
    stderr: _T
    def __init__(self, args: _CMD, returncode: int, stdout: _T | None = ..., stderr: _T | None = ...) -> None: ...
    def check_returncode(self) -> None: ...
    if sys.version_info >= (3, 9):
        def __class_getitem__(cls, item: Any) -> GenericAlias: ...

if sys.version_info >= (3, 7):
    # Nearly the same args as for 3.6, except for capture_output and text
    @overload
    def run(
        args: _CMD,
        bufsize: int = ...,
        executable: StrOrBytesPath | None = ...,
        stdin: _FILE = ...,
        stdout: _FILE = ...,
        stderr: _FILE = ...,
        preexec_fn: Callable[[], Any] | None = ...,
        close_fds: bool = ...,
        shell: bool = ...,
        cwd: StrOrBytesPath | None = ...,
        env: _ENV | None = ...,
        universal_newlines: bool = ...,
        startupinfo: Any = ...,
        creationflags: int = ...,
        restore_signals: bool = ...,
        start_new_session: bool = ...,
        pass_fds: Any = ...,
        *,
        capture_output: bool = ...,
        check: bool = ...,
        encoding: str | None = ...,
        errors: str | None = ...,
        input: str | None = ...,
        text: Literal[True],
        timeout: float | None = ...,
    ) -> CompletedProcess[str]: ...
```

Here the _ENV var is reported as unknown

**To Reproduce**
Example:
```
    subprocess.run(
       [
        'dpkg-buildpackage',
        '--no-sign'
       ],
       cwd=context.release_dir,
       check=True)
```

**Additional context**
This started with 1.1.250, it worked before.

Could the problem be that my pyrightconfig.json has set 
`"pythonPlatform": "All",` ?

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression introduced when I fixed [this bug](https://github.com/microsoft/pyright/issues/3511). 

The problem is that `_ENV` is declared twice in the stub. Normally only one of the declarations is considered because the other is considered "unreachable". But when you set "pythonPathform" to "All", both declarations are now visible to the type evaluator.

When multiple declarations are present, the type is evaluated as the union of both types. In the process of combining the two subtypes into a union, the symbol was no longer treated as a type alias because type aliases are not supposed to have multiple declarations.

I think the right behavior here is to detect the case where a symbol has multiple `TypeAlias` declarations and use only the first one, ignoring any subsequent ones for the purposes of type evaluation.

This change will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.251, which I just published. It will also be included in the next release of pylance.
